ee/server/service/software_installers.go

```diff
@@ -1850,8 +1884,37 @@ func (svc *Service) softwareBatchUpload(
 				installer.Filename = filename
 			}
 
+			if p.Slug != nil && *p.Slug != "" {
+				// Fleet maintained software hydration
+				// This code should be extracted for common use from here and AddFleetMaintainedApp in maintained_apps.go
+				// It's the same code and would be nice to get some reuse
+				appName := p.MaintainedApp.UniqueIdentifier
+				if p.MaintainedApp.Platform == "darwin" || appName == "" {
+					appName = p.MaintainedApp.Name
+				}
+				if installer.Filename == "" {
```

Comment:
is this going to work for apps like WhatsApp that don't have the file extension at the end of the download URL?

Reply:
That a great question. We have different ways of handling this in the `maintained_apps.go` code vs here.

Both really lean hard into the fact that we don't have an empty filename. But I actually copied over from https://github.com/fleetdm/fleet/blob/main/server/mdm/maintainedapps/installers.go#L76-L81 the empty file name approach. So thoughts on how to fix this would be much appreciated.

Reply:
@ksykulev ohh I see. I didn't realize that `DownloadInstaller` used this same method.

In that case, I think this is OK for now. This is a known issue that's outside scope for this ticket afaict.

Reply:
Maybe we can add in some handling for [this case](https://github.com/fleetdm/fleet/blob/main/server/mdm/maintainedapps/installers.go#L63-L73) here? Ideally via a shared function there in `server/mdm/maintainedapps`?